### ISSUE TITLE
GKE smoke tests don't match GKE cluster-up

### ISSUE THREAD

**Issue by deads2k** [kubernetes/kubernetes#29375]
GKE smoke tests passed for https://github.com/kubernetes/kubernetes/pull/28788, but the GKE tests broke with (see https://github.com/kubernetes/kubernetes/pull/29361)

```
23:23:00 Created [https://test-container.sandbox.googleapis.com/v1/projects/k8s-jkns-e2e-gke-ci/zones/us-central1-f/clusters/jenkins-e2e].
23:23:00 kubeconfig entry generated for jenkins-e2e.
23:23:00 NAME         ZONE           MASTER_VERSION                    MASTER_IP       MACHINE_TYPE   NODE_VERSION                      NUM_NODES  STATUS
23:23:00 jenkins-e2e  us-central1-f  1.4.0-alpha.1.564+8ead63f1270455  146.148.90.159  n1-standard-2  1.4.0-alpha.1.564+8ead63f1270455  3          RUNNING
23:23:00 ... calling validate-cluster
23:23:01 failed to find client for version v1: error: You must be logged in to the server (the server has asked for the client to provide credentials)
23:23:01 (kubectl failed, will retry 2 times)
23:23:02 failed to find client for version v1: error: You must be logged in to the server (the server has asked for the client to provide credentials)
23:23:02 (kubectl failed, will retry 1 times)
```

From what I understand, the actual scripts are internal, but @wojtek-t noted:

> @kubernetes/test-infra-maintainers @kubernetes/goog-gke - since it was failing in verify-cluster, it clearly means that GKE smoke tests are configured differently than all other suites. This should be fixed.

in https://github.com/kubernetes/kubernetes/pull/28788#issuecomment-234204179

@kubernetes/test-infra-maintainers @kubernetes/goog-gke

**Comment by deads2k** [comment]
@piosz cc (saw you on the original thread)
@kubernetes/rh-cluster-infra this is blocking a pull we need for security downstream (#28788) and I'm at loss as to how to debug it.

**Comment by piosz** [comment]
@deads2k thanks for filling this.

**Comment by cjcullen** [comment]
@spxtr for gke smoke tests. I'll dig into why #28788 broke GKE.

**Comment by wojtek-t** [comment]
Thanks @cjcullen !

**Comment by spxtr** [comment]
The GKE smoke test doesn't actually use kubernetes from your PR yet, it uses whatever is the default for GKE. I'll fix that in the next week.

**Comment by wojtek-t** [comment]
@spxtr - what's the status of it? I think we should prioritize it. 

@fejta - can you please prioritize it?

**Comment by wojtek-t** [comment]
To clarify - this is especially annoying for people outside Google, where they see that GKE tests are passing, and then we are reverting their PR because it broken GKE tests.

**Comment by spxtr** [comment]
Yes, this is a priority. It didn't happen this week because of the meetup but it will be first thing next week.
